### PR TITLE
add recovery policy and process state config

### PR DESCRIPTION
**Pull request by duoertai**

### CODE REVIEW

File: api-schema/xdb.yaml [hunk 1]
```diff
@@ -518,6 +524,11 @@ components:
         maximumAttemptsDurationSeconds: 
           type: integer
           description: the maximum duration of all attempts, default to 0, means unlimited
+    StateFailureRecoveryPolicy:
+      type: string
+      enum:
+        - FAIL_WORKFLOW_ON_STATE_FAILURE
```
Comment (longquanzheng) [inline api-schema/xdb.yaml:530]: FAIL_WORKFLOW -> FAIL_PROCESS

File: api-schema/xdb.yaml [hunk 2]
```diff
@@ -499,6 +499,12 @@ components:
           $ref: '#/components/schemas/RetryPolicy'
         executeApiRetryPolicy:
           $ref: '#/components/schemas/RetryPolicy'
+        stateFailureRecoveryPolicy:
```
Comment (longquanzheng) [inline api-schema/xdb.yaml:502]: WDYT we can put the whole thing under `stateFailureRecoveryPolicy`?
```
stateFailureRecoveryPolicy:
    stateFailureMode: FAIL_PROCESS_ON_STATE_FAILURE or   PROCEED_TO_CONFIGURED_STATE
    stateFailureProceedStateId: xxx  # for FAIL_PROCESS_ON_STATE_FAILURE only 
    stateFailureProceedStateConfig: ... # for FAIL_PROCESS_ON_STATE_FAILURE only
```

Comment (duoertai) [reply api-schema/xdb.yaml:511]: how about using oneof in state config, and define 2 types of recovery policy?

Comment (duoertai) [reply api-schema/xdb.yaml:512]: seems not supported, went with your idea

Comment (longquanzheng) [reply api-schema/xdb.yaml:512]: yeah, and it's not very well supported in different langauges. that's why I didn't use it at all.

File: api-schema/xdb.yaml [hunk 3]
```diff
@@ -411,6 +411,16 @@ components:
         attempt:
           description: for async state API only(during backoff retry)
           type: integer
+        lastFailedStateExecutionId:
```
Comment (longquanzheng) [inline api-schema/xdb.yaml:414]: nit: how about `recoverFromStateExecutionId` and `recoverFromStateApi`.

File: api-schema/xdb.yaml [hunk 4]
```diff
@@ -411,6 +411,17 @@ components:
         attempt:
           description: for async state API only(during backoff retry)
           type: integer
+        recoverFromStateExecutionId:
+          description: for async state API only, state id + sequence number
+          type: string
+        recoverFromApi:
+          description: if last execution failed, WAIT_UNTIL or EXECUTE, otherwise NONE
+          $ref: '#/components/schemas/ApiType'
+    ApiType:
```
Comment (longquanzheng) [inline api-schema/xdb.yaml:420]: Maybe State api type to Be more clear ?

**Review by longquanzheng — APPROVED** [review]

File: api-schema/xdb.yaml [hunk 5]
```diff
@@ -499,6 +510,22 @@ components:
           $ref: '#/components/schemas/RetryPolicy'
         executeApiRetryPolicy:
           $ref: '#/components/schemas/RetryPolicy'
+        stateFailureRecoveryInfo:
```
Comment (longquanzheng) [inline api-schema/xdb.yaml:513]: it maybe nice to rename this to `stateFailureRecoveryPolicy`

Comment (longquanzheng) [reply api-schema/xdb.yaml:513]: Also let's create a separate definition for `StateFailureRecoveryOptions` and use `$ref`

**Pull request merged**